src/llm/async_func.rs

```diff
@@ -108,16 +115,63 @@ impl AsyncFuncExpr {
             );
         };
 
-        async_udf
-            .invoke_async_with_args(
-                AsyncScalarFunctionArgs {
-                    args: args.to_vec(),
-                    number_rows: batch.num_rows(),
-                    schema: batch.schema(),
-                },
-                option,
-            )
-            .await
+        let mut result_batches = vec![];
+        if let Some(ideal_batch_size) = self.ideal_batch_size()? {
+            let mut remainder = batch.clone();
+            while remainder.num_rows() > 0 {
+                let current_batch = remainder.slice(0, ideal_batch_size); // get next 10 rows
+                remainder =
+                    remainder.slice(ideal_batch_size, remainder.num_rows() - ideal_batch_size);
+                let args = llm_function
+                    .args()
+                    .iter()
+                    .map(|e| e.evaluate(&current_batch))
+                    .collect::<Result<Vec<_>>>()?;
+                result_batches.push(
+                    async_udf
+                        .invoke_async_with_args(
+                            AsyncScalarFunctionArgs {
+                                args: args.to_vec(),
+                                number_rows: current_batch.num_rows(),
+                                schema: current_batch.schema(),
+                            },
+                            option,
+                        )
+                        .await?,
+                );
+            }
+        } else {
+            let args = llm_function
+                .args()
+                .iter()
+                .map(|e| e.evaluate(batch))
+                .collect::<Result<Vec<_>>>()?;
+
+            result_batches.push(
+                async_udf
+                    .invoke_async_with_args(
+                        AsyncScalarFunctionArgs {
+                            args: args.to_vec(),
+                            number_rows: batch.num_rows(),
+                            schema: batch.schema(),
+                        },
+                        option,
+                    )
+                    .await?,
+            );
+        }
+
+        let datas = result_batches
+            .iter()
+            .map(|b| b.to_data())
+            .collect::<Vec<_>>();
+        let total_len = datas.iter().map(|d| d.len()).sum();
+        let mut mutable = MutableArrayData::new(datas.iter().collect(), false, total_len);
+        datas.iter().enumerate().for_each(|(i, data)| {
+            mutable.extend(i, 0, data.len());
+        });
+        let array_ref = make_array(mutable.freeze());
```

Comment:
@alamb I tried merging the array results using `MutableArrayData.` Is this an efficient method?